### ISSUE TITLE
Plugin start-up triggers an iOS main thread violation

### ISSUE THREAD

**Issue by ettore** [newrelic/newrelic-capacitor-plugin#57]
[NOTE]: # ( ^^ Provide a general summary of the issue in the title above. ^^ )

We are developing an hybrid app using Capacitor for iOS (and Android) and recently integrated New Relic. We configured the node.js agent following the [NR docs here](https://docs.newrelic.com/docs/mobile-monitoring/new-relic-mobile-ionic-capacitor/get-started/introduction-new-relic-ionic-capacitor/#configure). The integration all happens on the web side. However when the plugin is started Xcode reports a main thread violation happening inside the NewRelicCapacitor plugin.

## Description

[NOTE]: # ( Describe the problem you're encountering. )
[TIP]:  # ( Do NOT give us access or passwords to your New Relic account or API keys! )

We start the plugin from our `main.tsx` file:
```
....
import { Capacitor } from "@capacitor/core";
import {
  AgentConfiguration,
  NREnums,
  NewRelicCapacitorPlugin,
} from "@newrelic/newrelic-capacitor-plugin";
....

const getAppToken = (): string | undefined => {
  ....
};

const setupNewRelic = () => {
  const appToken = getAppToken();

  if (!appToken) {
    console.info("New Relic app token not found");
    return;
  }

  let agentConfig: AgentConfiguration = {
    analyticsEventEnabled: true,
    webViewInstrumentation: true,
    crashReportingEnabled: true,
    interactionTracingEnabled: true,
    networkRequestEnabled: true,
    networkErrorRequestEnabled: true,
    httpResponseBodyCaptureEnabled: true,
    loggingEnabled: true,
    logLevel: NREnums.LogLevel.INFO,
    sendConsoleEvents: true,
  };

  NewRelicCapacitorPlugin.start({
    appKey: appToken,
    agentConfiguration: agentConfig,
  });
};

export default setupNewRelic;
.....
setupNewRelic();
```
On the native side this ends up in the [`NewRelicCapacitorPluginPlugin::start(:)`](https://github.com/newrelic/newrelic-capacitor-plugin/blob/632549fb6cc1537f6544e7f529335e5f8f37a3de/ios/Plugin/NewRelicCapacitorPluginPlugin.swift#L52) function, which at runtime may be called off of the main thread by Capacitor. Inside that function there's a [call](https://github.com/newrelic/newrelic-capacitor-plugin/blob/632549fb6cc1537f6544e7f529335e5f8f37a3de/ios/Plugin/NewRelicCapacitorPluginPlugin.swift#L161) to `NewRelic::start(withApplicationToken:)` function, which appears to use UIKit apis. Since this latter function is called outside of the main thread, this results in a main thread violation.

## Steps to Reproduce

[NOTE]: # ( Please be as specific as possible. )

Please see description.

## Expected Behavior

[NOTE]: # ( Tell us what you expected to happen. )

The Xcode main thread checker should not report any main thread violations, per Apple development guidelines.

## Relevant Logs / Console output

[NOTE]: # ( Please provide specifics of the local error logs, Browser Dev Tools console, etc. if appropriate and possible. )

Stack trace reported by Xcode:
```
=================================================================
Main Thread Checker: UI API called on a background thread: -[UIApplication applicationState]
PID: 51910, TID: 3443713, Thread name: (none), Queue name: bridge, QoS: 0
Backtrace:
4   NewRelic                            0x00000001050922d0 -[NewRelicAgentInternal initWithCollectorAddress:crashCollectorAddress:andApplicationToken:] + 168
5   NewRelic                            0x0000000105095690 __96+[NewRelicAgentInternal startWithApplicationToken:andCollectorAddress:andCrashCollectorAddress:]_block_invoke + 88
6   libdispatch.dylib                   0x0000000104731d50 _dispatch_client_callout + 16
7   libdispatch.dylib                   0x00000001047337f8 _dispatch_once_callout + 84
8   NewRelic                            0x000000010509562c +[NewRelicAgentInternal startWithApplicationToken:andCollectorAddress:andCrashCollectorAddress:] + 596
9   NewRelic                            0x00000001050953b4 +[NewRelicAgentInternal startWithApplicationToken:andCollectorAddress:] + 296
10  NewrelicNewrelicCapacitorPlugin     0x00000001044767a8 $s08NewrelicA15CapacitorPlugin08NewRelicbcC0C5startyySo13CAPPluginCallCF + 9356
11  NewrelicNewrelicCapacitorPlugin     0x0000000104476da4 $s08NewrelicA15CapacitorPlugin08NewRelicbcC0C5startyySo13CAPPluginCallCFTo + 52
12  Capacitor                           0x00000001044f8e08 $s9Capacitor0A6BridgeC12handleJSCall4callyAA0D0V_tFyycfU0_ + 916
13  Capacitor                           0x00000001044ecfd4 $sIeg_IeyB_TR + 48
14  libdispatch.dylib                   0x0000000104730528 _dispatch_call_block_and_release + 24
15  libdispatch.dylib                   0x0000000104731d50 _dispatch_client_callout + 16
16  libdispatch.dylib                   0x000000010473a014 _dispatch_lane_serial_drain + 924
17  libdispatch.dylib                   0x000000010473ad6c _dispatch_lane_invoke + 424
18  libdispatch.dylib                   0x0000000104748b74 _dispatch_workloop_worker_thread + 1716
19  libsystem_pthread.dylib             0x00000001b1834878 _pthread_wqthread + 284
20  libsystem_pthread.dylib             0x00000001b183363c start_wqthread + 8
```

## Your Environment

[TIP]:  # ( Include as many relevant details about your environment as possible. )
MBP with M1 cpu
MacOS 13.5 
Xcode 14.3.1 
iOS 16.4 simulator 
iOS 16.6.1 iPhone 11
React 18.2.0
New Relic node.js agent: 8.17.1
New Relic iOS agent: 7.4.6
New Relic Capacitor plugin: 1.3.1

## Additional context

[TIP]:  # ( Add any other context about the problem here. )
This appears to be fixed by simply wrapping the call to `NewRelic.start(withApplicationToken:)` in a common `DispatchQueue.main.async { }` block. If that's not violating other assumptions on NR or Capacitor side, I can submit a quick fix for this.

**Comment by ndesai-newrelic** [comment]
@ettore  That appears to be a viable solution. Could you please submit a PR (pull request) for it?

**Comment by ndesai-newrelic** [comment]
@ettore it is released now in 1.3.2

**Comment by ettore** [comment]
👍  thanks!
